### PR TITLE
config: support for UnmarshalYAML

### PR DESCRIPTION
**Pull request by mcorbin**

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -41,3 +43,25 @@ type Configuration struct {
...
+
+// UnmarshalYAML parses a reporter configuration from YAML.
+func (configuration *Configuration) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type rawConfiguration struct {
+		URL     string
+		Timeout uint64
+	}
+	// this is the default configuration
+	rawConfig := rawConfiguration{
+		URL:     "https://kroki.io",
```
Comment (ggrossetie) [inline config.go]: it should be `https://demo.kroki.io`:

```suggestion
		URL:     "https://demo.kroki.io",
```

Comment (mcorbin) [reply config.go]: Updated

File: config.go [hunk 2]
```diff
@@ -41,3 +43,25 @@ type Configuration struct {
 	URL     string
 	Timeout time.Duration
 }
+
+// UnmarshalYAML parses a reporter configuration from YAML.
```
Comment (ggrossetie) [inline config.go]: "a reporter configuration" ?

File: config.go [hunk 3]
```diff
@@ -41,3 +43,25 @@ type Configuration struct {
...
+func (configuration *Configuration) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
Comment (ggrossetie) [inline config.go:48]: It's the idiomatic way of doing it ? I'm a bit surprised that we do not return a Configuration or an error ?

File: config.go (hunk 2 above)
Comment (mcorbin) [reply config.go]: Updated :p

File: config.go (hunk 3 above)
Comment (mcorbin) [reply config.go:48]: Yes it it (cf https://godoc.org/gopkg.in/yaml.v2#Unmarshaler), it works by side effect (UnmarshalYAML receives a pointer to a struct and updates this struct).

Comment (ggrossetie) [reply config.go:48]: Thanks for the explanation 👍

**Review by ggrossetie — APPROVED** [review]
🚀

**Pull request merged**